### DISCUSSION TITLE
Zwave-js Library Compatibility to install in my ubuntu system

### DISCUSSION THREAD

**Discussion by Sabastian-Antony** [zwave-js/zwave-js#5563] [General]
Can this zwave-js library be installed and used on any system? Does it have any node/gyp version dependency? 

I am using node v10.15.3, npm 6.4.1, and node-gyp@8.1.0.
I am not able to successfully install the modules in my system

**Comment by AlCalzone** [comment]
Your node.js Version is too old. It needs at least 14.13 or 16.x or 18.x.

**Reply by Sabastian-Antony** [reply]
@AlCalzone Do we have any workaround to make it work?

**Reply by AlCalzone** [reply]
Update node.js?

**Reply by Sabastian-Antony** [reply]
@AlCalzone  I run node.js in an embedded system, where it is impossible to update the node due to resource constraints. So, if there is any workaround it will be helpful.

**Reply by AlCalzone** [reply]
You could try bundling `zwave-js` and all its dependencies using https://esbuild.github.io/ and have it [target](https://esbuild.github.io/api/#target) `node10`. I'm not sure if it will work though.

**Reply by kpine** [reply]
What system are you on where you can't upgrade Node? A manual [Node 16 install](https://nodejs.org/dist/latest-v16.x/) will run on Ubuntu 18.04, which ships older versions of node and npm (8.10.0 / 3.5.2). Depending on platform, the uncompressed install is probably 300-400MB (~351MB for ARM7).

If you're on such a resource constrained system, compiling and installing Z-Wave JS from source is probably harder than installing Node.

**Reply by Sabastian-Antony** [reply]
@kpine I use Debian GNU/Linux 8 (Jessie), so when I try to upgrade higher node version it throws glibc errors. So, I believe to resolve the glibc errors, ubuntu itself has to be upgraded which is not possible in our case.

**Reply by kpine** [reply]
What arch? Node 16 runs fine in Debian Jessie (as tested with `debian:eol/jessie` Docker image).

**Reply by Sabastian-Antony** [reply]
Our system is based on armv7l

**Reply by kpine** [reply]
I built and ran the latest ZUI with Node 16 in a Debian Jessie container without any noticeable issue. You can try following these instructions, replacing arm64 references with the armv7l equivalents. https://github.com/zwave-js/zwave-js-ui/issues/2973#issuecomment-1441354912

You will need 1-2GB of RAM (or swap) if you choose to build ZUI.
